projects/netgrif-components-core/src/lib/task/services/task-data.service.ts

```diff
@@ -413,41 +415,78 @@ export class TaskDataService extends TaskHandlingService implements OnDestroy {
 
         this._taskResourceService.setData(this._safeTask.stringId, body).pipe(take(1))
             .subscribe((response: EventOutcomeMessageResource) => {
-                this.processSuccessfulSetDataRequest(setTaskId, response.outcome, afterAction, nextEvent, body);
+                if (!this.isTaskRelevant(setTaskId)) {
+                    this._log.debug('current task changed before the set data response could be received, discarding...');
+                    this._taskState.stopLoading(setTaskId);
+                    this._taskState.stopUpdating(setTaskId);
+                    afterAction.complete();
+                    nextEvent.resolve(false);
+                    return;
+                }
+                if (response.success) {
+                    this.processSuccessfulSetDataRequest(setTaskId, response, afterAction, nextEvent, body);
+                } else if (response.error !== undefined) {
+                    this.processUnsuccessfulSetDataRequest(setTaskId, response, afterAction, nextEvent, body);
+                }
             }, error => {
                 this.processErroneousSetDataRequest(setTaskId, error, afterAction, nextEvent, body);
             });
     }
 
     /**
-     * Processes a successful outcome of a `setData` request
+     * Processes a unsuccessful outcome of a `setData` request
      * @param setTaskId the Id of the task whose data was set
      * @param response the resulting Event outcome of the set data request
      * @param afterAction the action that should be performed after the request is processed
      * @param nextEvent indicates to the event queue that the next event can be processed
      * @param body hold the data that was sent in request
      */
-    protected processSuccessfulSetDataRequest(setTaskId: string,
-                                              response: EventOutcome,
+    protected processUnsuccessfulSetDataRequest(setTaskId: string,
+                                              response: EventOutcomeMessageResource,
                                               afterAction: AfterAction,
                                               nextEvent: AfterAction,
                                               body: TaskSetDataRequestBody) {
-        if (!this.isTaskRelevant(setTaskId)) {
-            this._log.debug('current task changed before the set data response could be received, discarding...');
-            this._taskState.stopLoading(setTaskId);
-            this._taskState.stopUpdating(setTaskId);
-            afterAction.complete();
```

Comment:
it is okay that afterAction.complete and stop loading and updating was removed ?

Reply:
implement in performSetDataRequest

Reply:
its okey...